testrunner-running/src/main/scala/edu/illinois/cs/testrunner/testobjects/JUnit5TestClass.scala

```diff
@@ -8,7 +8,7 @@ import scala.collection.JavaConverters._
 class JUnit5TestClass(loader: ClassLoader, clz: Class[_]) extends GeneralTestClass {
 
     def fullyQualifiedName(method: Method): String =
-        clz.getCanonicalName ++ "#" ++ method.getName
+        clz.getName ++ "#" ++ method.getName
```

Comment:
Would this be a problem with JUnit 4 tests, i.e., do we need to do the same change in JUnitTestClass?

Reply:
Not sure. The test discovery process between J4 and J5 is different. I didn't check how J4 does that. And we don't even know if the current iDFlakies supports nested tests for J4 test.

Reply:
I think JUnit 4 can run nested test classes as long as they are public and static. In such a case, the class name should also use $, so getName should be preferred over getCanonicalName. Can you please also make the same change in JUnitTestClass?

Reply:
Sure

Reply:
Done